### PR TITLE
[docs] IA3

### PR DESCRIPTION
**Pull request by stevhliu**
Adds a guide for IA3 to the docs. The `IA3Config` is pretty barebones so feel free to suggest any improvements to it!

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
Thanks a lot !

**Comment by stevhliu** [comment]
Thanks! I'll wait for one more review from @BenjaminBossan when he returns from vacation before merging!

**Review by pacman100 — APPROVED** [review]
Thank you @stevhliu for adding a task guide on IA3 method! 🔥🚀✨

Left a few suggestions/comments.

File: docs/source/task_guides/ia3.md
Comment (BenjaminBossan) [inline docs/source/task_guides/ia3.md:36]: Let's add the import `from datasets import load_dataset`.

File: docs/source/task_guides/ia3.md
Comment (BenjaminBossan) [inline docs/source/task_guides/ia3.md:96]: Let's add the imports: 

```
from torch.utils.data import DataLoader
from transformers import default_data_collator
```

File: docs/source/task_guides/ia3.md
Comment (BenjaminBossan) [inline docs/source/task_guides/ia3.md:130]:
```suggestion
from peft import IA3Config, get_peft_model
```

File: docs/source/task_guides/ia3.md
Comment (BenjaminBossan) [inline docs/source/task_guides/ia3.md:143]: let's add the imports:

```
import torch
from transformers import get_linear_schedule_with_warmup
```

File: docs/source/task_guides/ia3.md
Comment (BenjaminBossan) [inline docs/source/task_guides/ia3.md:157]: This block misses `from tqdm import tqdm` and `device = "cuda"`.

File: docs/source/task_guides/ia3.md
Comment (BenjaminBossan) [inline docs/source/task_guides/ia3.md:211]: Should "stevhliu" be replaced here by `<your-hf-account-name>` for consistency with the snippet above?

**Review by BenjaminBossan — CHANGES_REQUESTED** [review]
Thanks a lot for adding the IA³ guide. Overall, it looks good. I would suggest to add just a couple of more lines of code to make it so that users can just copy and paste the code and it runs. WDYT?

**Review by BenjaminBossan — APPROVED** [review]
Thanks for making the adjustments quickly. LGTM.

**Pull request merged**
